scripts/gold_accounting.py

```diff
@@ -0,0 +1,346 @@
+
+    return user_creddits
+
+
+def run_for_prev_month():
+    end_date = datetime(datetime.now().year, datetime.now().month, 1)
+    if end_date.month == 1:
+        start_date = datetime(end_date.year-1, 12, 1)
+    else:
+        start_date = datetime(end_date.year, end_date.month-1, 1)
+
+    user_creddits = get_creddit_balances_on(start_date)
+
+    output = StringIO.StringIO()
+    writer = csv.writer(output, delimiter=',', quotechar='"')
+    data = [
+            'Date/time',
+            'Transaction ID',
+            'Type',
+            'Description',
+            'Amount Billed',
+            'Value',
+            'Processor',
+            'Period (Days)',
+            'Payer Account',
+            'Recipient Account']
+    writer.writerow(data)
+
+    transactions = GoldTransaction.get_transactions(start_date, end_date)
+    for trans in transactions:
+        value = trans.revenue
+
+        if trans.type == 'creddits':
+            creddit = Creddit(value=trans.revenue / trans.months,
+                              processor=trans.processor,
+                              purchase_time=trans.date)
+            user_creddits[trans.account_id].extend([creddit] * trans.months)
+            value = 0
+    
+        elif trans.type == 'giftcode':
+            value = 0
+
+        elif trans.type.endswith('-creddit'):
+            # if the user has creddits that we know how they paid for
+            num_creddits = len(user_creddits[trans.payer])
+            if num_creddits > 0:
+                value = sum(c.value for c in user_creddits[trans.payer])
+                value = value / num_creddits
+                # if this purchase would use more creddits than we know about
+                # make sure not to add value for non-existent creddits
+                value = value * min(num_creddits, trans.months)
+
+                # remove the used creddits
+                for _ in xrange(trans.months):
+                    try:
+                        user_creddits[trans.payer].popleft()
+                    except:
+                        break
+        data = [
+            trans.date,
+            trans.transaction_id,
+            "reversed" if trans.is_reversed else trans.type,
+            trans.description,
+            trans.revenue,
+            value,
+            trans.processor,
+            trans.days,
+            trans.payer,
+            trans.account_id
+        ]
+        writer.writerow(data)
+
+    body=("""<!doctype html><html><body>
+        <span>Attached is the requested gold accounting file for %s to %s</span>
+        </body></html>""" %
+         (start_date.strftime('%Y-%m-%d'), end_date.strftime('%Y-%m-%d')))
+ 
+    filename=('gold_accounting_{0}_to_{1}.csv'
+                .format(start_date.strftime('%Y-%m-%d'),
+                        end_date.strftime('%Y-%m-%d')))
+    send_email(body, [{"name": filename, "contents": output.getvalue()}])
+
+
+def send_email(body, attachment):
```

Comment:
nit: this should probably also be `attachments` just to make it clear it's a list.